db/features.go

```diff
@@ -0,0 +1,101 @@
+	errgo "gopkg.in/errgo.v1"
+
+	"github.com/Scalingo/cli/config"
+	"github.com/Scalingo/cli/io"
+	scalingo "github.com/Scalingo/go-scalingo/v6"
+)
+
+// EnableFeature is the command handler to enable a database feature on a given
+// database addon, like 'force-ssl' or 'public-availability'
+func EnableFeature(appctx *cli.Context, app, addon, feature string) error {
+	spinner := spinner.New(spinner.CharSets[11], 100*time.Millisecond)
+	spinner.Suffix = " Enabling database feature"
+	spinner.Start()
+	defer spinner.Stop()
+
+	client, err := config.ScalingoClient(appctx.Context)
+	if err != nil {
+		return errgo.Notef(err, "fail to get Scalingo client")
+	}
+
+	res, err := client.DatabaseEnableFeature(appctx.Context, app, addon, feature)
+	if err != nil {
+		return errgo.Notef(err, "fail to enable feature")
+	}
+	spinner.Stop()
+
+	switch res.Status {
+	case scalingo.DatabaseFeatureStatusActivated:
+		io.Statusf("Feature %v has been enabled\n", feature)
+	case scalingo.DatabaseFeatureStatusFailed:
+		io.Warningf("Feature %v failed to get activated, please contact our support\n", feature)
+	case scalingo.DatabaseFeatureStatusPending:
+		io.Statusf("Feature %v is being enabled\n", feature)
+	}
+
+	if res.Status == scalingo.DatabaseFeatureStatusPending && appctx.Bool("synchronous") {
+		io.Infof("Waiting for operation completion...")
+		err = waitFeatureUntilActivated(appctx.Context, client, app, addon, feature)
+		if err != nil {
+			return errgo.Notef(err, "fail to wait for feature to be enabled")
+		}
+	}
+
+	return nil
+}
+
+func waitFeatureUntilActivated(ctx context.Context, client *scalingo.Client, app, addon, feature string) error {
+	ticker := time.NewTicker(3 * time.Second)
+	defer ticker.Stop()
+	for range ticker.C {
+		db, err := client.DatabaseShow(ctx, app, addon)
+		if err != nil {
+			return errgo.Notef(err, "fail to refresh database metadata")
+		}
+		for _, f := range db.Features {
+			if f.Name == feature && f.Status != scalingo.DatabaseFeatureStatusPending {
+				switch f.Status {
+				case scalingo.DatabaseFeatureStatusActivated:
+					io.Statusf("Feature %v has been activated\n", feature)
+					return nil
+				case scalingo.DatabaseFeatureStatusFailed:
+					io.Warningf("Feature %v failed to get activated, please contact our support\n", feature)
+					return nil
+				}
+			}
+		}
+	}
+	return nil
+}
+
+// DisableFeature is the command handler to disable a database feature on a
+// database addon like 'force-ssl' or 'public-availability'
+func DisableFeature(ctx context.Context, app, addon, feature string) error {
+	spinner := spinner.New(spinner.CharSets[11], 100*time.Millisecond)
+	spinner.Suffix = " Disabling database feature"
+	spinner.Start()
+	defer spinner.Stop()
+
+	client, err := config.ScalingoClient(ctx)
+	if err != nil {
+		return errgo.Notef(err, "fail to get Scalingo client")
+	}
+
+	_, err = client.DatabaseDisableFeature(ctx, app, addon, feature)
+	if err != nil {
+		return errgo.Notef(err, "fail to disable feature")
```

Comment:
suggestion: maybe add the feature name in the error message?
```suggestion
return errgo.Notef(err, "fail to disable feature %v", feature)
```

Reply:
Ack